presto-parser/src/main/java/io/prestosql/sql/parser/ErrorHandler.java

```diff
@@ -273,41 +293,52 @@ else if (transition instanceof WildcardTransition) {
                             labels = labels.complement(IntervalSet.of(Token.MIN_USER_TOKEN_TYPE, atn.maxTokenType));
                         }
 
-                        if (labels.contains(currentToken)) {
-                            activeStates.push(new ParsingState(transition.target, tokenIndex + 1, false, caller, parser, current.depth + 1));
+                        if (labels.contains(currentToken) && tokenIndex < stream.size() - 1) {
```

Comment:
is `&& tokenIndex < stream.size() - 1` necessary?
i think EOF token serves as a sentinel.

Reply:
I had to add this in one of my earlier experiments, but you're right. Let me see if I can change it back.